src/rules/DeclarativeResourceRule.js

```diff
@@ -0,0 +1,69 @@
+/*
+ * DeclarativeResourceRule.js - subclass of ResourceRule that can iterate over
+ * an arrays of regular expressions to apply to a resource 
+ *
+ * Copyright © 2023 JEDLSoft
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import ResourceRule from './ResourceRule.js';
+
+class DeclarativeResourceRule extends ResourceRule {
+    /**
+     * Construct a new regular expression-based declarative resource rule.
+     *
+     * @param {Object} options options as documented above
+     * @constructor
+     */
+    constructor(options) {
+        super(options);
+
+        if (!options || !options.regexps) {
+            throw "Missing required options for the DeclarativeResourceRule constructor";
+        }
+        this.regexps = options?.regexps;
```

Comment:
Are we ever using these regexp source strings from options later? If not, we could just do
```js
this.re = options.regexps.map(/*etc*/)
```

Reply:
updated